### PR TITLE
Feature/filter by reviewer add issue comment

### PR DESCRIPTION
**Pull request by javad-amin**
Adds ability to just leave a comment on the pull request.
Ability to fetch the pull requests that you have reviewed.
Option to check the Github Action and have the status display for each pr. (It's kind of slow)
Tells you if the pull request needs to be rebased and basicly if it's ready to be merged.
Some refactoring and linting was done.

### CODE REVIEW

File: src/config.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    def get(
+        self: "GithubConfig",
+        key: str,
+        fallback: str = "",
+        section: str = DEFAULT_SECTION_NAME,
+    ) -> str:
+        return self.config.get(section, key, fallback=fallback)
+
+    def update(
+        self: "GithubConfig",
```
Comment (technomunk) [inline src/config.py]: **suggestion:** python 3.11 has a [special shortcut type](https://docs.python.org/3/library/typing.html?highlight=typing%20self#typing.Self) for the explicit purpose of typing self
```suggestion
from typing import Self

from configparser import ConfigParser

CONFIG_FILE_NAME = "config.ini"
DEFAULT_SECTION_NAME = "GitHub"


class GithubConfig:
    config = ConfigParser()

    def setup_config(self: Self) -> None:
        self.config.read(CONFIG_FILE_NAME)
        if DEFAULT_SECTION_NAME not in self.config.sections():
            self.config.add_section(DEFAULT_SECTION_NAME)

    def get(
        self: Self,
        key: str,
        fallback: str = "",
        section: str = DEFAULT_SECTION_NAME,
    ) -> str:
        return self.config.get(section, key, fallback=fallback)

    def update(
        self: Self,
```

File: src/pull_requests.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+@dataclass
+class SearchParameters:
+    org_name: str
+    review_requested_user: str
+    author: str
+    title: str
+    reviewed_by: str
+
+    def __post_init__(self: "SearchParameters") -> None:
```
Comment (technomunk) [inline src/pull_requests.py]:
```suggestion
from dataclasses import dataclass
from typing import Self

from github import Github
from github.PullRequest import PullRequest

from config import GithubConfig


@dataclass
class SearchParameters:
    org_name: str
    review_requested_user: str
    author: str
    title: str
    reviewed_by: str

    def __post_init__(self: Self) -> None:
```

File: src/app.py [hunk 3]
```diff
@@ -1,193 +1,25 @@
...
-    return pull_requests
-
-
-def setup_config() -> ConfigParser:
-    config = ConfigParser()
-    config.read(CONFIG_FILE)
-    if "GitHub" not in config.sections():
-        config.add_section("GitHub")
-    return config
+        pr_fetch_view(token)
```
Comment (technomunk) [inline src/app.py]: **praise:** good point to identify the opportunity to scope out the code and increase the separation of concerns 🌟

File: src/views/pull_requests.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+                mergability = f"{'✅ Mergable' if _is_ready_to_merge(pr) else '❌ Not Mergable'}"
+            else:
+                mergability = ""
+            needs_rebase = f"{' | ⚠️ Rebase required' if not pr.mergeable else ''}"
+            checked = st.checkbox(
+                f"{repo_name_link} | {pr.title} by {pr.user.login} | {mergability}{needs_rebase}",
+                value=select_all,
+            )
+
+            selection_result[pr] = checked
```
Comment (technomunk) [inline src/views/pull_requests.py]: **suggestion:** this forms looks like a prime candidate to become a helper function

File: src/config.py (hunk 1 above)
Comment (javad-amin) [reply src/config.py]: Great to know 🌟

**Review by technomunk — APPROVED** [review]
LGTM, this is a clear improvement. I want to experiment with parallelizing the flow so it would go quicker, but that's clearly a separate feature

**Comment by javad-amin** [comment]
> LGTM, this is a clear improvement. I want to experiment with parallelizing the flow so it would go quicker, but that's clearly a separate feature

Great idea! I still have some issues/bugs but I'll leave them to a different day.

**Pull request merged**